docs/docs/python/integrations/aws-lambda.mdx

```diff
@@ -66,6 +65,40 @@ Use [layout](../../../reference/targets/python_aws_lambda_function.mdx#layout) t
 `file` / `files` targets will not be included in the built AWS Lambda artifacts because filesystem APIs like `open()` would not load them as expected. Instead, use the `resource` and `resources` target. See [Assets and archives](../../using-pants/assets-and-archives.mdx) for further explanation.
 :::
 
+### Specifying a runtime explicitly
+
+Pants will attempt to infer the appropriate runtime and its platform-specific details from [the relevant interpreter constraints](../overview/interpreter-compatibility.mdx). For example, given `interpreter_constaints = ["==3.12.*"]`, Pants will build the Lambda package to appropriate for the  `python3.12` runtime. The Lambda environment is a relatively stable, running Linux on either a x86-64 or arm64 CPU, and thus Pants includes default "complete platforms" specifications that allow choosing appropriate artifacts for third-party dependencies with native code. These are chosen based on the (inferred or explicit) runtime and [the architecture](../../../reference/targets/python_aws_lambda_function.mdx#architecture).
```

Comment:
> I think this is actually a bit misleading because the architecture defaults to x86 if not specified

In my mind the architecture is still being used when choosing in this case, just the value being used is from the default rather than a default

> runtime is preferred over interpreter constraints

Yeah, fair enough. I'll tweak.